src/components/Misc/PostButtonModal.tsx

```diff
@@ -0,0 +1,40 @@
+import React, { Dispatch, ReactElement, SetStateAction } from "react";
+import { Dialog, DialogTitle, DialogContent, Button } from "@mui/material/";
+
+const styles = {
+  dialogPosition: {
+    position: "absolute",
+    bottom: "500px",
+  },
+};
+
+type ModalProps = {
+  children: ReactElement<any, any>;
+  openModal: boolean;
+  setOpenModal: Dispatch<SetStateAction<boolean>>;
```

Comment:
This type looks really gross. I think it's better to pass this as a callback function that gets invoked. We can then call it `onClose` or something similar and pass that to the modal to set the close state. The type for that would be a function signature like this:
```suggestion
  onClose: () => void;
```

In the parent component we'd pass in the prop like so:

`<PostButtonModal onClose={() => setOpenModal (false)}>`

Reply:
I have applied this change.